docs/assignment3.md

```diff
@@ -1 +1,709 @@
 # Assignment 3
+
```

Comment:
Very well structured, well thought out and designed project. The code is very clean and the diagrams are clear and compact. The explanations are also very clear and understandable. Congratulations! :)